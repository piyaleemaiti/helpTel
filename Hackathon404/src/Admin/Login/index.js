import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  TextInput,
  StatusBar,
  SafeAreaView,
  Button,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import styles from './style';
import commonStyle from '../../commonStyles';

export default function Login({navigation}) {
  const [stateData, setStateData] = useState({
    emailId: '',
    password: '',
    hotelName: '',
    hotelId: 0,
    listOfServices: [],
    emailIdError: false,
    passwordError: false,
  });

  useEffect(() => {}, []);
  const onChangeText = (text, name) => {
    const nameField = `${name}Error`;
    if (text === '') {
      setStateData(state => ({
        ...state,
        [nameField]: true,
      }));
    } else {
      setStateData(state => ({
        ...state,
        [nameField]: false,
      }));
    }
    setStateData(state => ({...state, [name]: text}));
  };
  const login = () => {
    if (stateData.emailId === '') {
      setStateData(state => ({
        ...state,
        emailIdError: true,
      }));
    } else if (stateData.password === '') {
      setStateData(state => ({
        ...state,
        passwordError: true,
      }));
    } else {
      setStateData(state => ({
        ...state,
        emailIdError: false,
        passwordError: false,
      }));

      fetch('https://backendproject5.herokuapp.com/fetchHotelDetails')
        .then(result => result.json())
        .then(lists => {
          lists.forEach(list => {
            if (list.hotelId === 1234) {
              navigation.navigate('AdminDashboard', {hotelId: list.hotelId});
            }
          });
        })
        .catch(err => console.log(err));
    }
  };
  const backToPage = () => {
    navigation.goBack();
  };
  return (
    <View style={commonStyle.container}>
      <StatusBar barStyle="dark-content" backgroundColor="skyblue" />
      <SafeAreaView>
        <View style={commonStyle.header}>
          <Icon
            name="arrow-left"
            style={commonStyle.backButton}
            size={20}
            onPress={backToPage}
          />
          <Text style={commonStyle.heading}>Login</Text>
        </View>
        <View style={commonStyle.content}>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Email address/Username: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'emailId')}
              value={stateData.emailId}
              placeholder="Email Address/User name"
            />
            {stateData.emailIdError && (
              <Text style={styles.errorMsg}>Please enter email address</Text>
            )}
          </View>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Password: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'password')}
              value={stateData.password}
              placeholder="Password"
              secureTextEntry={true}
            />
            {stateData.passwordError && (
              <Text style={styles.errorMsg}>Please enter password address</Text>
            )}
          </View>
          <Button
            title="Login"
            onPress={() => login()}
            style={{paddingTop: 10}}
          />
        </View>
      </SafeAreaView>
    </View>
  );
}
