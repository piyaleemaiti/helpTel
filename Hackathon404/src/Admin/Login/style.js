import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    padding: 5,
  },
  inputTitle: {
    fontSize: 16,
    paddingBottom: 10,
  },
  hotelName: {
    textAlign: 'center',
    fontSize: 20,
    fontWeight: '400',
    paddingBottom: 30,
    color: '#0B5FA4',
  },
  emailContainer: {
    paddingBottom: 20,
  },
  errorMsg: {
    color: 'red',
  },
});

export default styles;
