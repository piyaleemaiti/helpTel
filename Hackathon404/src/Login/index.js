import React, {useState, useEffect} from 'react';
import {getFetch, postFetch} from '../utils/fetchAPI';
import LoginComponent from './loginComponent';

export default function Login({navigation}) {
  const [stateData, setStateData] = useState({
    emailId: '',
    password: '',
    hotelName: '',
    hotelId: 0,
    listOfServices: [],
    passwordError: false,
    emailIdError: false,
    loginError: false,
  });

  useEffect(() => {
    getFetch('http://backendproject5.herokuapp.com/fetchHotelDetails')
      .then(lists => {
        console.log(lists);
        lists.forEach(list => {
          if (list.hotelId === 1234) {
            setStateData(state => ({
              ...state,
              hotelName: list.hotemName,
              hotelId: list.hotelId,
              listOfServices: list.listOfServices,
            }));
          }
        });
      })
      .catch(err => console.log(err));
  }, []);

  const onChangeText = (text, name) => {
    const nameField = `${name}Error`;
    if (text === '') {
      setStateData(state => ({
        ...state,
        [nameField]: true,
      }));
    } else {
      setStateData(state => ({
        ...state,
        [nameField]: false,
      }));
    }
    setStateData(state => ({...state, [name]: text}));
  };

  const _login = () => {
    if (stateData.emailId === '') {
      setStateData(state => ({
        ...state,
        emailIdError: true,
      }));
    } else if (stateData.password === '') {
      setStateData(state => ({
        ...state,
        passwordError: true,
      }));
    } else {
      const body = {
        userId: stateData.emailId,
        password: stateData.password,
      };

      postFetch(
        'http://backendproject5.herokuapp.com/fetchUserDetailsById',
        body,
      ).then(response => {
        console.log('response', response);
        if (response.status === 'Failed') {
          setStateData(state => ({
            ...state,
            loginError: true,
          }));
        } else {
          switch (response.userType) {
            case 'HOTEL_ADMIN':
              navigation.navigate('AdminDashboard', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
            case 'GUEST':
              navigation.navigate('Dashborad', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
            case 'SUPER_ADMIN':
              navigation.navigate('Dashborad', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
          }
        }
      });
    }
  };
  const backToPage = () => {
    navigation.goBack();
  };
  return (
    <LoginComponent
      onChangeText={onChangeText}
      backToPage={backToPage}
      _login={_login}
      emailIdError={stateData.emailIdError}
      passwordError={stateData.passwordError}
      loginError={stateData.loginError}
      emailId={setStateData.emailId}
      password={stateData.password}
    />
  );
}
