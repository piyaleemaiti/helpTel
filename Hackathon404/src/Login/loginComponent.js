import React from 'react';
import {
  View,
  Text,
  TextInput,
  StatusBar,
  SafeAreaView,
  Button,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import styles from './style';
import commonStyle from '../commonStyles';

const LoginComponent = props => {
  const {
    onChangeText,
    emailIdError,
    passwordError,
    loginError,
    backToPage,
    emailId,
    password,
    _login,
  } = props;

  return (
    <View style={commonStyle.container}>
      <StatusBar barStyle="dark-content" backgroundColor="skyblue" />
      <SafeAreaView>
        <View style={commonStyle.header}>
          <Icon
            name="arrow-left"
            style={commonStyle.backButton}
            size={20}
            onPress={backToPage}
          />
          <Text style={commonStyle.heading}>Login</Text>
        </View>
        <View style={commonStyle.content}>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Email address/Username: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'emailId')}
              value={emailId}
              placeholder="Email Address/User name"
            />
            {emailIdError && (
              <Text style={styles.errorMsg}>Please enter email address</Text>
            )}
          </View>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Password: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'password')}
              value={password}
              placeholder="Password"
              secureTextEntry={true}
            />
            {passwordError && (
              <Text style={styles.errorMsg}>Please enter password address</Text>
            )}
          </View>
          {loginError && (
            <Text style={styles.errorMsg}>
              Please enter valid email address/password
            </Text>
          )}
          <Button
            title="Login"
            onPress={() => _login()}
            style={{paddingTop: 10}}
          />
        </View>
      </SafeAreaView>
    </View>
  );
};

export default LoginComponent;
