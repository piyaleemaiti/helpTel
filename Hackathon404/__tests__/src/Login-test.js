/**
 * @format
 */

import 'react-native';
import React from 'react';
import Login from '../../src/Login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
fetch = jest.fn(() => Promise.resolve());
const navigation = {navigate: jest.fn()};

it('renders correctly', async () => {
  const container = renderer.create(<Login navigation={navigation} />);
  expect(container).toMatchSnapshot();
});
