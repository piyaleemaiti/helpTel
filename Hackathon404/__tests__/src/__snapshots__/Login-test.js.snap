// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "top": 40,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "skyblue",
          "flexDirection": "row",
          "height": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
              "paddingLeft": 10,
              "paddingRight": 20,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        ÔÅ†
      </Text>
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "paddingBottom": 10,
            }
          }
        >
          Email address/Username: 
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email Address/User name"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "paddingBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "paddingBottom": 10,
            }
          }
        >
          Password: 
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
